marketplacetests/marketplace/pages/base.py

```diff
@@ -36,11 +35,13 @@ def login(self, username, password):
         self.wait_for_login_success_notification()
 
     @property
-    def first_free_app_name(self):
+    def first_free_app(self):
         free_apps = self.search(':free').search_results
-        name = free_apps[0].name
-        self.tap_back()
-        return name
+        app = free_apps[0]
+        name = app.name
+        author = app.author
+        self.tap_home()
+        return {'name': name, 'author': author}
```

Comment:
The problem is that the call to `self.tap_home()` causes the variable `app` to lose its context, so we cannot call `app.name` after that, which is why I stored the values in a variable. I guess I could just create the dictionary on line 41 and then return that on line 44, which would save a line of code. I'll do that.